pkg/sources/gitlab/gitlab.go

```diff
@@ -252,22 +256,24 @@ func (s *Source) getRepos() ([]*url.URL, []error) {
 				continue
 			}
 
-			// The repo normalization has already successfully parsed the URL at this point, so we can ignore the error.
-			u, _ := url.ParseRequestURI(repo)
-			validRepos = append(validRepos, u)
+			validRepos = append(validRepos, repo)
 		}
 		return validRepos, errs
 	}
 	return nil, nil
 
 }
 
-func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk, repos []*url.URL) []error {
+func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk) []error {
 	wg := sync.WaitGroup{}
 	var errs []error
 	var errsMut sync.Mutex
 
-	for i, u := range repos {
+	// If there is resume information available, limit this scan to only the repos that still need scanning.
+	reposToScan, progressIndexOffset := sources.FilterReposToResume(s.repos, s.GetProgress().EncodedResumeInfo)
+	s.repos = reposToScan
```

Comment:
**Question:** Is it okay to be possibly dropping some information here?

Reply:
Are you referring to the dropping of the repositories that have already been scanned?

If so, I think it's fine. We don't actually store that information anywhere, as it's generated at the start of a scan. And the progressIndexOffset will allow us to get the proper count of the original number of repos.

However if you're referring to the potential to drop repos that may have been added between the time the scan started and when it was picked back up, I think that is okay. The goal here is to reduce the amount of time it takes to finish the scan, so the next scan will happen faster, and that scan will pick up the new repository.